cfpb/consumerfinance.gov
# Add documents report

Adds a report with details of all the documents managed in Wagtail. See GHE/CFGOV/platform/issues/4339 for more info.

---




## Additions

- A documents report using [Wagtail's standard reports interface](https://docs.wagtail.org/en/stable/extending/adding_reports.html)
- Unit tests for the new documents report

## Changes

- Since this is our second custom report in Wagtail, I made the unit tests—including the name of the test file—a little more generic to cover all the reports

## How to test this PR

1. Load the latest dump of production data
2. Go to `/admin/reports/documents/`, where you should see a list of all the documents managed in Wagtail
3. Hit the download CSV button on that list to make sure the CSV download works and all the data looks good in the CSV

## Notes and todos

- I'm going to add line comments once I open this PR to a couple of things I wasn't quite sure about that I'm hoping you lovely reviewers can pay extra attention to.

## Checklist

  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance
